tests/unit/test_functioncalls.py

```diff
@@ -0,0 +1,79 @@
+import numpy as np
+from unittest import TestCase
+import micro_manager
+
+
+class MicroSimulation:
+    def __init__(self):
+        self.very_important_value = 0
+
+    def initialize(self):
+        pass
+
+    def solve(self, macro_data, dt):
+        assert macro_data["macro-scalar-data"] == 1
+        assert macro_data["macro-vector-data"].tolist() == [0, 1, 2]
+        return {"micro-scalar-data": macro_data["macro-scalar-data"] + 1,
+                "micro-vector-data": macro_data["macro-vector-data"] + 1}
+
+
+class TestFunctioncalls(TestCase):
+    def setUp(self):
+        self.fake_read_data_names = {"macro-scalar-data": False, "macro-vector-data": True}
+        self.fake_read_data = [{"macro-scalar-data": 1, "macro-vector-data": np.array([0, 1, 2])}] * 4
+        self.fake_write_data_names = {"micro-scalar-data": False, "micro-vector-data": True}
+        self.fake_write_data = [{"micro-scalar-data": 1,
+                                 "micro-vector-data": np.array([0, 1, 2]), "micro_sim_time": 0}] * 4
+        self.macro_bounds = [0.0, 25.0, 0.0, 25.0, 0.0, 25.0]
+
+    def test_micromanager_constructor(self):
+        manager = micro_manager.MicroManager('test_unit.json')
+        self.assertListEqual(manager._macro_bounds, self.macro_bounds)
+        self.assertDictEqual(manager._read_data_names, self.fake_read_data_names)
+        self.assertDictEqual(dict(self.fake_write_data_names, **{'micro_sim_time': False}), manager._write_data_names)
+        self.assertEqual(manager._micro_n_out, 10)
+
+    def test_initialize(self):
+        manager = micro_manager.MicroManager('test_unit.json')
+        manager.initialize()
+        self.assertEqual(manager._dt, 0.1)  # from Interface.initialize
+        self.assertEqual(manager._global_number_of_micro_sims, 4)
+        self.assertListEqual(manager._macro_bounds, self.macro_bounds)
+        self.assertListEqual(manager._mesh_vertex_ids.tolist(), [0, 1, 2, 3])
+        self.assertEqual(len(manager._micro_sims), 4)
+        self.assertEqual(manager._micro_sims[0].very_important_value, 0)  # test inheritance
+        self.assertDictEqual(manager._read_data_names, self.fake_read_data_names)
+        self.assertDictEqual(dict(self.fake_write_data_names, **{'micro_sim_time': False}), manager._write_data_names)
+
+    def test_read_write_data_from_precice(self):
+        manager = micro_manager.MicroManager('test_unit.json')
+        manager.initialize()
+        manager.write_data_to_precice(self.fake_write_data)
+        read_data = manager.read_data_from_precice()
+        for data, fake_data in zip(read_data, self.fake_write_data):
+            self.assertEqual(data["macro-scalar-data"], 1)
+            self.assertListEqual(data["macro-vector-data"].tolist(),
+                                 fake_data["micro-vector-data"].tolist())
+
+    def test_solve_mico_sims(self):
+        manager = micro_manager.MicroManager('test_unit.json')
+        manager.initialize()
```

Comment:
Same as above.

Reply:
We seem to be needing `initialize` for solve_micro_sims see [here](https://github.com/precice/micro-manager/actions/runs/5044114144/jobs/9046745186?pr=37#step:5:25). Should we just set the local number of micro_sims by hand?